tasks/tsconfig.json

```diff
@@ -1,25 +1,43 @@
 {
   "compilerOptions": {
-    "target": "es5",
-    "lib": ["dom", "dom.iterable", "esnext"],
+    "target": "esnext",
+    "module": "esnext",
+    "lib": [
+      "esnext",
+      "dom",
+      "dom.iterable"
+    ],
     "allowJs": true,
     "skipLibCheck": true,
     "strict": true,
-    "forceConsistentCasingInFileNames": true,
     "noEmit": true,
     "esModuleInterop": true,
-    "module": "esnext",
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
     "resolveJsonModule": true,
     "isolatedModules": true,
     "jsx": "preserve",
-    "incremental": true
+    "incremental": true,
+    "forceConsistentCasingInFileNames": true,
+    "allowSyntheticDefaultImports": true,
+    // "noPropertyAccessFromIndexSignature": true, // TODO: reenable later (didn't know that tsconfig.json supported comments lol)
```

Comment:
`noPropertyAccessFromIndexSignature` sounds good! When is later?

Reply:
In a follow up clean up PR, don't want to change even more in a single PR 😅 
There are quite a few places where this causes errors, I'll try to fix them and then turn on the setting. Currently in the process of doing a lot of clean up already; will see if I can reasonably fit these changes in with that PR